main.yml

```
@@ -400,6 +400,44 @@ parameters:
           # Pull in the APPUiO managed info labels
           * on(cluster_id) group_left(sales_order) appuio_managed_info{
               cloud_provider=~"%(cloud_provider)s",
+              distribution=~"openshift4",
+              vshn_service_level="%(vshn_service_level)s",
+          }
+
+      appuio_managed_cores:
+        enabled: true
+        prometheus_org_id: appuio-managed-openshift-billing
+        products:
+          - product_id: 'openshiftoke-baremetal-core-besteffort'
+            params:
+              cloud_provider: baremetal
+              vshn_service_level: best_effort
+          - product_id: 'openshiftoke-baremetal-core-guaranteed'
+            params:
+              cloud_provider: baremetal
+              vshn_service_level: guaranteed_availability
+        instance_id_pattern: 'ocp-cores-%(cluster_id)s-%(role)s'
+        item_group_description_pattern: 'APPUiO Managed OpenShift - Cluster: %(cluster_id)s'
+        unit_id: 'uom_uom_198_02905845' # Core-Hour
+        query_pattern: |
+          # Calculates cores for app nodes of a cluster
+          # Max values over one hour.
+          max_over_time(
+            # Sum the vCPUs by cluster
+            sum by(cluster_id, role) (
+                # Get the node vCPUs
+                node_cpu_info
```

This counts hyperthreads and not cores, you'll want the following to actually count cores (as far as I understand we bill cores and not hyperthreads):

```suggestion
                max without (cpu) (node_cpu_info)
```

---

TIL